omod/src/test/java/org/openmrs/module/reportingrest/web/resource/ReportRequestResourceTest.java

```diff
@@ -60,7 +63,39 @@ public void setUp() throws Exception {
 	private ReportRequestResource getResource() {
 		return (ReportRequestResource) Context.getService(RestService.class).getResourceBySupportedClass(ReportRequest.class);
 	}
-	
+
+    @Test
+    public void testCreate() throws Exception {
+        String reportDefinitionJson = "{"
+                + "\"parameterizable\":{"
+                + "   \"uuid\":\"" + REPORT_DEFINITION_UUID + "\""
+                + "},"
+                + "\"parameterMappings\":{"
+                + "   \"endDate\":\"2017-01-31\","
+                + "   \"startDate\":\"2017-01-01\"}"
+                + "}";
+
+
+        String reportRequestJson = "{\n" +
+                "  \"status\": \"REQUESTED\",\n" +
+                "  \"priority\": \"NORMAL\",\n" +
+                "  \"reportDefinition\":" + reportDefinitionJson + "," +
+                "  \"renderingMode\": \"org.openmrs.module.reporting.report.renderer.CsvReportRenderer\"\n" +
+                "}";
+
+        SimpleObject properties = SimpleObject.parseJson(reportRequestJson);
+
+        RequestContext context = new RequestContext();
+        context.setRepresentation(Representation.DEFAULT);
+        SimpleObject response = (SimpleObject) getResource().create(properties, context);
+        assertNotNull(response.get("uuid"));
+        assertEquals(response.get("status"), ReportRequest.Status.REQUESTED);
+        assertEquals(response.get("priority"), ReportRequest.Priority.NORMAL);
+        SimpleObject resultObject = (SimpleObject) response.get("renderingMode");
+        String rendererType = (String) resultObject.get("rendererType");
```

Comment:
This doesn't seem to be testing the Mapped<ReportDefinition> is coming back correctly.  Is this tested and found to work?

Reply:
yes @mseaton, this test case is not covering the different Mapped types. I will update in the next PR.